From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: FirebirdTA01 <FirebirdTA01@aol.com>
Date: Sun, 18 Oct 2026 23:12:44 -0500
Subject: [PATCH 37/38] spu: target builtins read DECL_MD_FUNCTION_CODE (GCC
 10)

spu_resolve_overloaded_builtin and spu_expand_builtin are only called for
BUILT_IN_MD functions; DECL_FUNCTION_CODE is the BUILT_IN_NORMAL accessor
and asserts that class in a checking build.
---
 gcc/config/spu/spu-c.cc | 2 +-
 gcc/config/spu/spu.cc   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/gcc/config/spu/spu-c.cc b/gcc/config/spu/spu-c.cc
index be39de89e72..e4c8f12035b 100644
--- a/gcc/config/spu/spu-c.cc
+++ b/gcc/config/spu/spu-c.cc
@@ -94,7 +94,7 @@ spu_resolve_overloaded_builtin (location_t loc, tree fndecl, void *passed_args,
 			  || POINTER_TYPE_P (t))
   vec<tree, va_gc> *fnargs = static_cast <vec<tree, va_gc> *> (passed_args);
   unsigned int nargs = vec_safe_length (fnargs);
-  int new_fcode, fcode = DECL_FUNCTION_CODE (fndecl);
+  int new_fcode, fcode = DECL_MD_FUNCTION_CODE (fndecl);
   struct spu_builtin_description *desc;
   tree match = NULL_TREE;
 
diff --git a/gcc/config/spu/spu.cc b/gcc/config/spu/spu.cc
index a8c0d90d9d1..693524ae5c0 100644
--- a/gcc/config/spu/spu.cc
+++ b/gcc/config/spu/spu.cc
@@ -6645,7 +6645,7 @@ spu_expand_builtin (tree exp,
 		    int ignore ATTRIBUTE_UNUSED)
 {
   tree fndecl = TREE_OPERAND (CALL_EXPR_FN (exp), 0);
-  unsigned int fcode = DECL_FUNCTION_CODE (fndecl);
+  unsigned int fcode = DECL_MD_FUNCTION_CODE (fndecl);
   struct spu_builtin_description *d;
 
   if (fcode < NUM_SPU_BUILTINS)
